### ISSUE TITLE
Paystack popup flow complaining of duplicate transaction ref

### ISSUE THREAD

**Issue by mannyAndem** [a11rew/medusa-payment-paystack#74]
Hi, When I try to use the popup flow as described in the README, The paystack popup throws an error of duplicate transaction reference. I'm guessing its because the plugin creates a transaction already but I'm not 100% sure. Is this the case, and how can I get around this? Thanks in advance!

**Comment by a11rew** [comment]
@mannyAndem yes the plugin creates a transaction on the backend you "resume" on the frontend by using the transaction reference added to the session data. 

If by "getting around this" you mean providing your own transaction reference, that's not supported by the plugin. It needs the transaction reference to verify the transaction on the backend. What's your use case for providing your own transaction reference?

**Comment by mannyAndem** [comment]
I don't want to use my own transaction reference. The problem I'm running into is that while using the pop-up flow recommended by the docs, the pop-up attempts to create a new transaction with the ref I pass. Please how exactly do I resume an already existing transaction ?

**Comment by a11rew** [comment]
Gotcha. You don't have to do anything special to resume a transaction, that's the default behaviour when you provide a reference tied to an existing payment. 

Outside of setup mistakes like passing an email, amount and currency to the popup different from what the plugin sends to Paystack on your backend, I could only see this happening from trying to retry a settled transaction. i.e. A transaction marked by Paystack as completed/errored/abandoned. 

If so, you should could either intitialise a new transaction (take users to the cart page again) or manually [refresh the payment session](https://docs.medusajs.com/api/store#carts_postcartscartpaymentsessionssession). 

You'd need to provide a reproduction or a lot more info for further help.

**Comment by mannyAndem** [comment]
Thank you soo much! It appears the issue was caused by me passing the wrong amount into the popup's setup. I assumed the amount was in naira and passed in amount  * 100 instead of just the amount itself.

Everything works well now, really appreciate the time.

I did notice however, that the docs still uses the v1 of the paystack popup which has been deprecated. Could I perhaps help in updating that section of the docs?
